src/main.c

```diff
@@ -504,7 +481,7 @@ verify(void *userdata, const rustls_verify_server_cert_params *params) {
     }
   }
   fprintf(stderr, "ocsp response len: %ld\n", params->ocsp_response.len);
-  if(0 != strcmp((const char *)userdata, "verify_arg")) {
+  if(0 != strcmp(((struct demo_conn*)userdata)->verify_arg, "verify_arg")) {
```

Comment:
nit: three opening parens in a row suggests you could move the cast to `struct demo_conn *` out of the `if` condition for clarity.